File: CSharp/Test/MyMessage.cs
using Protoserial;

namespace Test
{
    [Protoserial.Message]
    class MyMessage : ParentMessage
    {
        public int? varIntegerNullable;
        public ushort varUshort;
        public short varShort;
        public uint varUint;
        public int varInteger;
        public ulong varUlong;
        public long varLong;
        public float varFloat;
        public double varDouble;
        public string varString;
        public OtherMessage varOther;

        [Required] public string requiredField;
    }
}

File: Java/Test/src/MyMessage.java
import com.conditionracer.protoserial.*;

@Message
public class MyMessage extends ParentMessage
{
    public Integer varIntegerNullable;
    @Unsigned public Short varUshort;
    public Short varShort;
    @Unsigned public Integer varUint;
    public Integer varInteger;
    @Unsigned public Long varUlong;
    public Long varLong;
    public Float varFloat;
    public Double varDouble;
    public String varString;
    public OtherMessage varOther;

    public Integer[] varRepeatedInt;
    public OtherMessage[] varRepeatedOther;

    @Required public String requiredField;
}
